l1-contracts/src/core/Rollup.sol

```diff
@@ -213,23 +221,66 @@ contract Rollup is Leonidas, IRollup {
       revert Errors.Rollup__InvalidProposedArchive(expectedArchive, _archive);
     }
 
-    bytes32[] memory publicInputs =
-      new bytes32[](4 + Constants.HEADER_LENGTH + Constants.AGGREGATION_OBJECT_LENGTH);
-    // the archive tree root
-    publicInputs[0] = _archive;
+    // TODO(#7346): Currently verifying block root proofs until batch rollups fully integrated.
+    // Hence the below pub inputs are BlockRootOrBlockMergePublicInputs, which are larger than
+    // the planned set (RootRollupPublicInputs), for the interim.
+    // Public inputs are not fully verified (TODO(#7373))
+
+    bytes32[] memory publicInputs = new bytes32[](
+      Constants.BLOCK_ROOT_OR_BLOCK_MERGE_PUBLIC_INPUTS_LENGTH + Constants.AGGREGATION_OBJECT_LENGTH
+    );
+
+    // From block_root_or_block_merge_public_inputs.nr: BlockRootOrBlockMergePublicInputs.
+    // previous_archive.root: the previous archive tree root
+    publicInputs[0] = expectedLastArchive;
+    // previous_archive.next_available_leaf_index: the previous archive next available index
+    publicInputs[1] = bytes32(header.globalVariables.blockNumber);
+
+    // new_archive.root: the new archive tree root
+    publicInputs[2] = expectedArchive;
     // this is the _next_ available leaf in the archive tree
     // normally this should be equal to the block number (since leaves are 0-indexed and blocks 1-indexed)
     // but in yarn-project/merkle-tree/src/new_tree.ts we prefill the tree so that block N is in leaf N
-    publicInputs[1] = bytes32(header.globalVariables.blockNumber + 1);
-
-    publicInputs[2] = vkTreeRoot;
-
-    bytes32[] memory headerFields = HeaderLib.toFields(header);
-    for (uint256 i = 0; i < headerFields.length; i++) {
-      publicInputs[i + 3] = headerFields[i];
+    // new_archive.next_available_leaf_index: the new archive next available index
+    publicInputs[3] = bytes32(header.globalVariables.blockNumber + 1);
+
+    // TODO(#7346): Currently previous block hash is unchecked, but will be checked in batch rollup (block merge -> root).
+    // block-building-helpers.ts is injecting as 0 for now, replicating here.
+    // previous_block_hash: the block hash just preceding this block (will eventually become the end_block_hash of the prev batch)
+    publicInputs[4] = bytes32(0);
+
+    // TODO(#7346): Move archive membership proof to contract?
+    // verifyMembership(archivePath, _previousBlockHash, header.globalVariables.blockNumber - 1, expectedLastArchive)
+
+    // end_block_hash: the current block hash (will eventually become the hash of the final block proven in a batch)
+    publicInputs[5] = _currentBlockHash;
+
+    // TODO(#7346): Move archive membership proof to contract?
+    // Currently archive root is updated by adding the new block hash inside block-root circuit.
+    // verifyMembership(archivePath, _currentBlockHash, header.globalVariables.blockNumber, expectedArchive)
```

Comment:
This is pretty messy (sorry) for a few reasons:
- As commented in a few places in the code, it would be a big change to extract and input the `previousBlockHash` for a block-root circuit (it's not used now, but required for batch rollups, but the prover doesn't 'know' yet about blocks that came before it AFAIK). For now, I've set it to 0 and it originates in `block-building-helpers`. When merge-root is used, it will check that `left.end_block_hash == right.previous_block_hash`. The very first `previous_block_hash` will be bubbled up to the final root where it will be checked on L1, but maybe the other checks (archive, block numbers, etc.) are sufficient?
- The plan (as in [this doc](https://github.com/AztecProtocol/engineering-designs/blob/main/in-progress/7346-batch-proving-circuits-and-l1.md)) is to move archive membership to L1. In this PR, membership is assured as block hashes are added to the (pedersen) archive tree in the block-root circuit to find the new archive root, so for now adding a membership check would be an unnecessary and high gas cost change.

Reply:
We are currently storing the `archive` in the `BlockLog` to support non-sequential proving, and as we needed it as input to the proof validation anyway. 

If we are going the "TxObjects" direction (meeting Thursday) we won't have the archive at that point, and it will instead make sense for us to store the `BlockHash` as we need something to sign over for the committee anyway.

With that change, you don't need the membership check here as it would simply be reading the values and checking directly against those instead.

Shortterm, I think a fine approach to get the logic down would be to extend `BlockLog` with the hash, and then you can simple read those or perform the check similarly to the archive checks.

Reply:
Thanks for the suggestion, have added `blockHash` to `BlockLog`!